packages/asas-lookups/src/asas_lookups/router.py

```
@@ -80,9 +81,15 @@ def list_values(
         type_ = service.get_type(session, type_key)
         # Cheap revalidation: ETag keyed on the type version (bumped on any change)
         # and the caller's org (WXL-241 — two orgs see different value sets of one
-        # type).
+        # type). The query shape must be in the tag too — an ETag is per
+        # *representation*, and page 2 or a filtered list is a different body
+        # than page 1: without this a conforming client 304-reuses page 1 for
+        # every other page/filter of the same type version.
         org = service._current_org(session)
-        etag = f'W/"{type_.key}.v{type_.version}.{lang}.o{org or 0}"'
+        shape = hashlib.sha1(
```

Ruff (0.16.2)

[error] Probable use of insecure hash functions in `hashlib`: `sha1`

(S324)